DSharpPlus/Utilities.cs

```diff
@@ -289,6 +431,33 @@ public static bool Contains(this string str, params char[] characters)
             return false;
         }
 
+        /// <summary>
+        /// An async version of Regex.Replace
+        /// </summary>
+        /// <param name="regex">The Regex object which is used to match the string for replacements</param>
+        /// <param name="input">The string to search for a match</param>
+        /// <param name="replacementFn">A custom method that examines each match and returns either the original matched string or a replacement string</param>
+        /// <returns>A new string that is identical to the input string, except that a replacement
+        /// string takes the place of each matched string. If pattern is not matched in the
+        /// current instance, the method returns the current instance unchanged</returns>
+        public static async Task<string> ReplaceAsync(this Regex regex, string input, Func<Match, Task<string>> replacementFn)
```

Comment:
What is the use case of this

Reply:
That's used inside the humanization method, just because I needed to call async methods while regexping. It could be made private, I kept it public only because of similar methods around the class.